pkg/providers/amifamily/suite_test.go

```diff
@@ -303,6 +303,180 @@ var _ = Describe("AMIProvider", func() {
 			}))
 		})
 	})
+	Context("AMI List requirements", func() {
+		BeforeEach(func() {
+			nodeClass.Spec.AMISelectorTerms = []v1.AMISelectorTerm{
+				{
+					Tags: map[string]string{"*": "*"},
+				},
+			}
+		})
+		It("should priortize the older non-deprecated ami without deprecation time", func() {
+			// Here we have two AMIs one which is deprecated and newer and one which is older and non-deprecated without a deprecation time
+			// List operation will priortize the non-deprecated AMI without the deprecation time
+			awsEnv.EC2API.DescribeImagesOutput.Set(&ec2.DescribeImagesOutput{
+				Images: []*ec2.Image{
+					{
+						Name:            aws.String(amd64AMI),
+						ImageId:         aws.String("ami-5678"),
+						CreationDate:    aws.String("2021-08-31T00:12:42.000Z"),
+						DeprecationTime: aws.String(time.Now().Add(-1 * time.Minute).Format(time.RFC3339)),
```

Comment:
same here on not using a minute so it's not racey
```suggestion
						DeprecationTime: aws.String(time.Now().Add(-1 * time.Hour).Format(time.RFC3339)),
```